tests/pact/fqns.repl

```diff
@@ -77,3 +77,165 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+; In the following tests, we define a module `test-mod-redeploy-ref`, and then
+; redeploy the same module with the change to one capability: `test`.
+; In the old version, the `test` capability fails, in the new one it passes.
+
+(begin-tx)
+; First, demonstrate the behavior prior to pact-4.8.
+(env-exec-config ["DisablePact48"])
+
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+; Before pact-4.8, the updated capability will be ignored, and calls to a function
+; requiring that capability will fail. This is true whether we invoke `f` with or
+; without its module name, and with or without its namespace.
+(expect-failure "Execution expected to fail due to assertion failure \
+                \within defcap invoke - invoking with an unqualified reference." (f))
+
+(expect-failure "Execution expected to fail due to assertion failure \
+                \within defcap invoke - invoking with an qualified reference \
+                \(no namespace)." (test-mod-redeploy-ref.f))
+
+(expect-failure "Execution expected to fail due to assertion failure within \
+                \defcap invoke - invoking with an fully qualified reference."
+                (free.test-mod-redeploy-ref.f))
+(commit-tx)
+
+; The following module redeployment changed the capability `test` to pass.
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+  (defcap test ()
+    true)
+
+  (defun f ()
+    (with-capability (free.test-mod-redeploy-ref.test)
+      1))
+
+  (defun f1 ()
+    (with-capability (test-mod-redeploy-ref.test)
+      1))
+
+  )
+; Before pact-4.8, the capability update (passing `test`) was ignored as the
+; full-qualified reference referenced the previously deployed version of the module.
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
```

Comment:
Continuing on the above, suggest simply "Reproduce upgrade resolution bug with fully-qualified reference."